harness/determined/debug.py

```diff
@@ -0,0 +1,137 @@
+import contextlib
```

Comment:
non-blocking: what do you think about moving the profile stuff in this file to a separate file called `profile.py`, since profiling doesn't necessarily have to be debug related.

Reply:
I moved them to `det.experimental.profile` and `det.experimental.debug` with strong warnings against depending on their APIs.  I think these could be good starting points for eventually having some good public apis for debugging and profiling models.